### PR TITLE
Add retry policy

### PR DESCRIPTION
**Pull request by dmerrick**
c.p. https://github.com/argoproj-labs/hera-workflows/pull/116

More info: https://argoproj.github.io/argo-workflows/retries/#retry-policies

### CODE REVIEW

**Comment by dmerrick** [comment]
@flaviuvadan would you mind taking a look at this?

File: src/hera/retry.py [hunk 1]
```diff
@@ -17,8 +19,9 @@ class Retry(BaseModel):
     """
 
     duration: Optional[int]
-    max_duration: Optional[int]
```
Comment (flaviuvadan) [inline src/hera/retry.py:20]: Any chance you're open to adjusting the docstring to reflect the available fields? 🙂

File: src/hera/retry_policy.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+class RetryPolicy(str, Enum):
+    """A representation of the retry policy for a task.
+
+    Use retryPolicy to choose which failures to retry:
+
+      * Always: Retry all failed steps
+      * OnFailure: Retry steps whose main container is marked as failed in Kubernetes
+      * OnError: Retry steps that encounter Argo controller errors, or whose init or wait containers fail
+      * OnTransientError: Retry steps that encounter errors defined as transient, or errors matching the TRANSIENT_ERROR_PATTERN environment variable. Available in version 3.0 and later.
```
Comment (flaviuvadan) [inline src/hera/retry_policy.py:12]: This is great. Thanks for adding the explanations for the error types 🙂

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "hera-workflows"
-version = "3.0.0"
+version = "3.0.1"
```
Comment (flaviuvadan) [inline pyproject.toml:3]: Any chance this can be 3.1.0?

**Review by flaviuvadan — COMMENTED** [review]
This looks great, @dmerrick 🙂 I left questions about docs and the version. Once they're addressed I will merge it right away.

**Review by flaviuvadan — APPROVED** [review]
Thank you for the contribution @dmerrick! 🚀 going to merge post-build.

**Pull request merged**